internal/validation/gateway/httproute.go

```diff
@@ -39,11 +61,10 @@ func ValidateHTTPRoute(httproute *gatewayv1beta1.HTTPRoute, attachedGateways ...
 		}
 	}
 
-	// validate that no unsupported features are in use
-	if err := validateHTTPRouteFeatures(httproute); err != nil {
-		return false, "httproute spec did not pass validation", err
+	// Validate by sending converted routes to validation endpoint of Kong Gateway.
+	if routesSvc != nil {
+		return validateWithKongGateway(ctx, routesSvc, parserFeatures, kongVersion, httproute)
```

Comment:
I don't have an immediate solution for this but I'm not a huge fan of an implicit dependency like the `routesSvc` param 🤔 

Perhaps it would be better to create dedicated type(s) which could have a specific purpose of validating against Kong Gateway or not. 🤔

Reply:
Yup, when I wrote this I thought to myself it's not great, but also not terrible, see what I'm proposing now, using small no-op validator